theano/tensor/signal/downsample.py

```diff
@@ -518,6 +600,15 @@ def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max'):
                 " 'average_inc_pad' and 'average_exc_pad'. Got %s" % mode)
         self.mode = mode
 
+    def infer_shape(self, node, in_shapes):
+        return [in_shapes[0]]
+
+
+class MaxPoolGrad(PoolGrad):
```

Comment:
I think we need to be able to reload old pickled graph. I talked with @lamblin about this and this is the best way we found to do this:

DownsampleFactorMaxGradGrad = MaxPoolGrad

And allow in \__init__ and grad all the mode in that class, but not perform (add an assert in perform and c code to not compute bad value).

then make a quick optimization that would convert MaxPoolGrad to AveragePoolGrad. Register it in the canonizer phase.

Do you need more details? I forgot if you already did an optimization, the description is long, but the code is short.

thanks
bellow this class.